libxrdp/xrdp_rdp.c

```diff
@@ -160,9 +161,43 @@ xrdp_rdp_read_config(struct xrdp_client_info *client_info)
                 client_info->use_fast_path = 0;
             }
         }
-        else if (g_strcasecmp(item, "disableSSLv3") == 0)
+        else if (g_strcasecmp(item, "ssl_protocols") == 0)
         {
-            client_info->disableSSLv3 = g_text2bool(value);
+            char s[strlen(value) + 2];
+            snprintf(s, sizeof(s), "%s%s", value, "x");
+
+            /* disable all protocols first, enable later */
+            client_info->ssl_protocols =
+                SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2;
+
+            if (g_pos(value, "TLSv1.2") >= 0)
+            {
+                log_message(LOG_LEVEL_DEBUG, "TLSv1.2 enabled");
+                client_info->ssl_protocols &= ~SSL_OP_NO_TLSv1_2;
+            }
+            if (g_pos(value, "TLSv1.1") >= 0)
+            {
+                log_message(LOG_LEVEL_DEBUG, "TLSv1.1 enabled");
+                client_info->ssl_protocols &= ~SSL_OP_NO_TLSv1_1;
+            }
+            if (g_pos(value, "TLSv1 ") >= 0 || g_pos(s, "TLSv1x") >= 0)
```

Comment:
I want to keep these version strings as same as constants defined in `openssl/ssl.h`

```
# define SSL_TXT_SSLV2           "SSLv2"
# define SSL_TXT_SSLV3           "SSLv3"
# define SSL_TXT_TLSV1           "TLSv1"
# define SSL_TXT_TLSV1_1         "TLSv1.1"
# define SSL_TXT_TLSV1_2         "TLSv1.2"
```

I tested it actually matches.  I did a hack.  In case of no trailing space after `TLSv1`,  append "x" to the end and then compare to "TLSv1x" like a technique `test "x$var" = "xyes"` in shell script.  It is hacky but actually working.

I don't intend to merge this before v0.9.1.